### PR TITLE
lint: upgrade eslint to version 9

### PR DESCRIPTION
**Pull request by CGQAQ**
## Changes

- Upgrade eslint to version 9

### CODE REVIEW

**Review by ematipico — APPROVED** [review]

**Comment by CGQAQ** [comment]
> and I seem to get some errors from built files
I did not reproduce that, but I'll add `"**/dist/**"` anyway

image

File: eslint.config.js [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+export default [
+	// If ignores is used without any other keys in the configuration object, then the patterns act as global ignores.
+	// ref: https://eslint.org/docs/latest/use/configure/configuration-files#globally-ignoring-files-with-ignores
+	{
+		ignores: [
+			"**/.*",
+			"**/*.d.ts",
+			"packages/**/*.min.js",
+			"packages/**/dist/",
```
Comment (CGQAQ) [inline eslint.config.js:34]: It's already there. @bluwy

Comment (CGQAQ) [reply eslint.config.js:34]: And I did not reproduce that

Comment (CGQAQ) [reply eslint.config.js:34]: The CI is also passing, what env are you in?

Comment (bluwy) [reply eslint.config.js:34]: Ah it's the `benchmark/dist/` that's triggering it. I think it's also possible for directories like `examples/blog` to have dist when you're testing things locally. So would be nice to put a blanket ignore for `dist`.

Comment (CGQAQ) [reply eslint.config.js:34]: Got it

Comment (CGQAQ) [reply eslint.config.js:34]: `pnpm build && pnpm benchmark` will reproduce it

Comment (CGQAQ) [reply eslint.config.js:34]: fixed

**Comment by CGQAQ** [comment]
@bluwy PTAL

**Review by bluwy — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**